src/main/java/io/vertx/core/net/TrustOptions.java

```diff
@@ -44,4 +45,25 @@
   default TrustManagerFactory getTrustManagerFactory(Vertx vertx) throws Exception {
     return KeyStoreHelper.create((VertxInternal) vertx, this).getTrustMgrFactory((VertxInternal) vertx);
   }
+
+  /**
+   * Returns a function that maps SNI server names to a {@link TrustManagerFactory} instance.
+   *
+   * The returned {@code TrustManagerFactory} must already be initialized and ready to use.
+   *
+   * The mapper is only used when the server has SNI enabled and the client indicated a server name.
+   * <p/>
+   * The returned function may return {@code null} in which case {@link #getTrustManagerFactory(Vertx)} is used as fallback.
+   *
+   * @param vertx the vertx instance
+   * @return the trustManager
+   */
+  default Function<String, TrustManagerFactory> trustManagerMapper(Vertx vertx) throws Exception {
```

Comment:
we should rather return null when not found and let the SSLHelper use the default value like it is for the `KeyCertOptions`

Reply:
given that the SSLHelper actually seems to do that